### PR TITLE
argtable3 3.2.2 (new formula)

### PR DESCRIPTION
**Pull request by domrim**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This library is required to build [blisp](https://github.com/pine64/blisp) from tarball.

### CODE REVIEW

File: Formula/a/argtable3.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  url "https://github.com/argtable/argtable3/archive/refs/tags/v3.2.2.f25c624.tar.gz"
+  sha256 "a5c66d819fa0be0435f37ed2fb3f23e371091722ff74219de97b65f6b9914e51"
+  license "BSD-3-Clause"
+
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", "-DBUILD_SHARED_LIBS=ON", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "ctest", "--test-dir", "build"
```
Comment (p-linnane) [inline Formula/a/argtable3.rb:13]:
```suggestion

```

File: Formula/a/argtable3.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        assert (arg_nullcheck(argtable) == 0);
+        if (arg_parse(argc, argv, argtable) == 0) {
+          if (all->count) puts ("Received option");
+        } else {
+          puts ("Invalid option");
+        }
+      }
+    EOS
+    system ENV.cc, "test.c", "-L#{lib}", "-I#{include}", "-largtable3",
```
Comment (p-linnane) [inline Formula/a/argtable3.rb:36]:
```suggestion

    system ENV.cc, "test.c", "-L#{lib}", "-I#{include}", "-largtable3",
```

File: Formula/a/argtable3.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+class Argtable3 < Formula
+  desc "ANSI C library for parsing GNU-style command-line options"
+  homepage "https://www.argtable.org"
+  url "https://github.com/argtable/argtable3/archive/refs/tags/v3.2.2.f25c624.tar.gz"
+  sha256 "a5c66d819fa0be0435f37ed2fb3f23e371091722ff74219de97b65f6b9914e51"
+  license "BSD-3-Clause"
```
Comment (p-linnane) [inline Formula/a/argtable3.rb:6]:
```suggestion
  license "BSD-3-Clause"
  head "https://github.com/argtable/argtable3.git", branch: "master"

```

**Review by p-linnane — COMMENTED** [review]
Thanks for the PR. Just a few small nits and this should be good to merge.

File: Formula/a/argtable3.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+class Argtable3 < Formula
+  desc "ANSI C library for parsing GNU-style command-line options"
+  homepage "https://www.argtable.org"
+  url "https://github.com/argtable/argtable3/archive/refs/tags/v3.2.2.f25c624.tar.gz"
```
Comment (ZhongRuoyu) [inline Formula/a/argtable3.rb:4]: Version number scanning [incorrectly returns `624`](https://github.com/Homebrew/homebrew-core/actions/runs/11196858337/job/31126187386?pr=192923#step:3:43) so we need to override it:

```suggestion
  url "https://github.com/argtable/argtable3/archive/refs/tags/v3.2.2.f25c624.tar.gz"
  version "3.2.2"
```

File: Formula/a/argtable3.rb
Comment (chenrui333) [inline Formula/a/argtable3.rb:1]: looks like argtable or argtable is not maintained, maybe a good idea to have that as versioned formula and move this formula to use argtable instead?

Comment (domrim) [reply Formula/a/argtable3.rb:1]: i'm not very familiar with the project... there are still some formulas which depend on argtable and argtable v2 and v3 are not api-compatible

Comment (chenrui333) [reply Formula/a/argtable3.rb:1]: yeah, I guess we can go with argtable3 (I was thinking argtable3 was successor to argtable project)

Comment (chenrui333) [reply Formula/a/argtable3.rb:1]: actually it got covered in this thread, https://github.com/argtable/argtable3/issues/58

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@domrim, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**